.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+on: [push]
+
+jobs:
+  checks:
+    name: Code Checks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v2
+      - name: Generate Dependency Updates Report
+        run: ./gradlew clean dependencyUpdates
+      - name: Code formatting
+        run: ./gradlew spotlessCheck
+      - name: PMD
+        run: ./gradlew pmdBenchmark pmdIntegrationTest pmdMain pmdTest
+      - name: Spotbugs
+        run: ./gradlew spotbugsBenchmark spotbugsIntegrationTest spotbugsMain spotbugsTest
+
+  unit_tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v2
+      - name: Unit Tests
+        run: ./gradlew test
+
+  integration_tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: serializing workflow runs
+        uses: softprops/turnstyle@v1
+        with:
+          same-branch-only: false
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Build with Gradle
+        run: ./gradlew clean setLibraryVersion integrationTest test jacocoTestCoverageVerification jacocoTestReport
```

Comment:
I'm just curious: Is ``test`` task required here, does it run unit-test again?

Reply:
The test task was also defines in the travis build. @ahmetoz can you clarify why we need the 'test'-task here

Reply:
good question, maybe we need to remove the test here and check the coverage report, if it works as expected then it's ok to remove test.

Reply:
Looks like it's needed, what do you think about merging everything in one step like in Travis, so when code check is not working it will not run unit tests and when unit tests are not working it would not run integration tests? Running them in parallel and making them faster might be handled in another task? WDYT?

Reply:
yes lets merge it in one step.... we can improve it later if needed

Reply:
6a517ef